src/core/execute.c

```diff
@@ -2576,12 +2583,6 @@ static int exec_child(
                 }
         }
 
-        r = apply_working_directory(context, params, home, needs_mount_namespace);
-        if (r < 0) {
-                *exit_status = EXIT_CHROOT;
-                return r;
-        }
-
```

Comment:
Hm, I read the commit message, and I see what is happening, but I don't understand *why*.